src/controllers/confirm.company.controller.ts

```diff
@@ -17,10 +22,33 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
+  const session: Session = req.session as Session;
+  const companyNumber = req.query["companyNumber"] as string;
+  try {
+    const eligibilityStatusCode: EligibilityStatusCode = await checkEligibility(session, companyNumber);
+    if (eligibilityStatusCode === EligibilityStatusCode.COMPANY_VALID_FOR_SERVICE) {
+      await createNewConfirmationStatement(req);
+      const url = TRADING_STATUS_PATH.replace(":companyNumber", companyNumber);
+      return res.redirect(url);
+    } else {
+      return displayEligibilityStopPage(res, eligibilityStatusCode);
+    }
+  } catch (e) {
+    return next(e);
+  }
+};
+
+const displayEligibilityStopPage = (res: Response, eligibilityStatusCode: EligibilityStatusCode) => {
+  if (eligibilityStatusCode === EligibilityStatusCode.INVALID_COMPANY_STATUS) {
+    return res.render(Templates.INVALID_COMPANY_STATUS);
+  }
+  throw new Error("Unknown eligibilityStatusCode");
```

Comment:
can you include the actual code like 
throw new Error(`Unknown eligibilityStatusCode ${eligibilityStatusCode}`);
might need to use eligibilityStatusCode.toString, just make sure it prints out the code correctly in the logs